### ISSUE TITLE
[$500] [MEDIUM] mweb - In employee WS, opening an expense detail page and tapping back, page gets scrolled.

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#34573]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.25-4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Tap on employee Workspace chat with lot of expenses
3. Tap on any IOU to open detail page
4. Tap back button and note page gets scrolled
5. Tap back button again and note page gets scrolled

## Expected Result:
When in employee WS, opening an expense detail page and tapping back, page must not get scrolled.

## Actual Result:
When in employee WS, opening an expense detail page and tapping back, page gets scrolled.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/e46302a2-9ac7-467f-bef4-c4f2449d2847

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01122bfdd943c30cad</li>
        <li>Upwork Job ID: 1747243693339979776</li>
        <li>Last Price Increase: 2024-05-07</li>
    </ul>
</details>

**Comment by Ykumar1415** [comment]
I tried to produce this issue and it it showing the same.

**Comment by situchan** [comment]
No proposals yet

**Comment by CortneyOfstad** [comment]
Not overdue

**Comment by CortneyOfstad** [comment]
Not overdue...

**Comment by situchan** [comment]
Still looking for proposals

**Comment by CortneyOfstad** [comment]
Not overdue – still waiting on proposals.

**Comment by CortneyOfstad** [comment]
@kbecciv can we confirm if the issue is still taking place?

**Comment by kbecciv** [comment]
We cannot check the issue, because when user is tapping back button, user navigated to LHN.

https://github.com/Expensify/App/assets/93399543/4d02c0e2-ca54-485c-93bc-589442e59b43

**Comment by CortneyOfstad** [comment]
@situchan correct me if I am wrong, but based on @kbecciv's comment [here](https://github.com/Expensify/App/issues/34573#issuecomment-1927888321), it is working as intended now? I feel like it is but want a buddy check. Thanks!

**Comment by situchan** [comment]
After ideal nav, there's navigation bug - all back button press goes to LHN whatever current page is.
So not able to test for now. I think we should hold this for https://github.com/Expensify/App/issues/35626

**Comment by CortneyOfstad** [comment]
Sounds good @situchan — updated the title and changed frequency 👍

**Comment by CortneyOfstad** [comment]
It shows that the on-hold issue is scheduled for payment on 2/20 (pending any regressions) so will keep an eye on that 👍

**Comment by CortneyOfstad** [comment]
Alright, @situchan the on-hold issue has been resolved — can you confirm if you're able to test? And if so, can you provide an update by EOD tomorrow? Thank you!

**Comment by situchan** [comment]
yes still reproducible



https://github.com/Expensify/App/assets/108292595/53fffee6-bff6-41e6-9d49-c0be450efd09

**Comment by CortneyOfstad** [comment]
Thank you @situchan!

**Comment by CortneyOfstad** [comment]
Per the comment [here](https://expensify.enterprise.slack.com/archives/C01GTK53T8Q/p1708964561503999), going to add this to Wave 6. Puneet confirmed this would be "medium" so adjusted the frequency 👍

**Comment by CortneyOfstad** [comment]
Reaching out to Callstack [here](https://expensify.slack.com/archives/C03UK30EA1Z/p1708966586763429).

**Comment by hussnainhamid1** [comment]
@CortneyOfstad Are we still accepting proposals here?

**Comment by hussnainhamid1** [comment]
Contributor details
Your Expensify account email: hussnainhamid50.hh9@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~012ccb6332580700af

**Comment by CortneyOfstad** [comment]
@situchan can you provide any feedback on @hussnainhamid1 proposal [here](https://github.com/Expensify/App/issues/34573#issuecomment-1964648140) by EOD? Thanks!

**Comment by situchan** [comment]
I don't see any proposal from @hussnainhamid1

**Comment by CortneyOfstad** [comment]
Weird — the comment must have been removed 🤔 

@hussnainhamid1 can you provide your proposal here for this issue?

**Comment by CortneyOfstad** [comment]
Hi @JmillsExpensify! I am heading OoO and will be back March 11th, so reassigned to have someone keep this moving until I am back. At this stage, we're waiting for @hussnainhamid1 to restate is proposal, otherwise this is still open to contributors. Thanks!

**Comment by situchan** [comment]
Still awaiting proposals

**Comment by Piotrfj** [comment]
Hi, Im Piotr from Callstack, I will start investigation in this issue.

**Comment by Piotrfj** [comment]
@kbecciv was it tested on real device? Android emulator is kinda slow by itself and it may cause this issue. I will try to test it on some real device and see how its going.

**Comment by kbecciv** [comment]
@Piotrfj Yes, QA team is not using emulator - only real devices.

**Comment by TomaszFrackowiak** [comment]
@JmillsExpensify - Hey, just to let you know, @Piotrfj is on sick leave, he will be back on Thursday

**Comment by CortneyOfstad** [comment]
Back from OoO — thanks for holding down the fort @JmillsExpensify!

Also, thanks for the heads up @TomaszFrackowiak!

**Comment by CortneyOfstad** [comment]
Not overdue as @Piotrfj should be back today 👍

**Comment by Piotrfj** [comment]
Yes, I'm back from sick leave. Working on it. Expecting to give some meaningful info today or tomorrow.

**Comment by CortneyOfstad** [comment]
Welcome back @Piotrfj and sounds good! Any questions along the way, just let me or @situchan know 👍

**Comment by Piotrfj** [comment]
I can see that this is still as FlatList and is waiting for refactoring to FlashList. If the issue isn't so crucial I would suggest to wait for the refactoring because this might solve the issue by itself.

**Comment by CortneyOfstad** [comment]
@Piotrfj thanks for the update! Coming from a non-engineering perspective, when you mention "refactoring," is there another GH or post that is focused on this that I can reference? I'm unfamiliar with this process, so would love to have someone tangible to reference/keep an eye on to avoid this getting lost in the shuffle of GHs.

**Comment by Piotrfj** [comment]
@CortneyOfstad from what I see this refactoring is related to this component
https://github.com/Expensify/App/issues/33725

**Comment by situchan** [comment]
The holding issue is also on hold.
@roryabraham do you think we should fix this issue urgently or wait for FlashList migration?

**Comment by roryabraham** [comment]
I think we're a long, long way away from migrating the main chat list to FlashList, because it lacks support for bidirectional pagination

**Comment by Piotrfj** [comment]
@roryabraham alright, I will continue the work on this issue as soon as I will end my current bug.

**Comment by DylanDylann** [comment]
I can take over this issue as a C+ reviewer because @situchan is OOO in April

**Comment by CortneyOfstad** [comment]
Is @situchan out of office already? Also, how long will they be OoO?

If they are only going to be gone a short while (couple of days), we can wait for them to return since this is a "polish" level GH. However, if they are going to be gone for a couple of weeks, then we can look into reassignment.

**Comment by mallenexpensify** [comment]
@situchan is out OOO, reassigning a C+ via the auto-assigner for fairness.

**Comment by CortneyOfstad** [comment]
Sounds good — thanks @mallenexpensify!

**Comment by CortneyOfstad** [comment]
@Piotrfj can you provide an update on the status of this by EOD today? Thanks!

**Comment by Piotrfj** [comment]
@CortneyOfstad Sure, I've found recently that the scrolling is working correctly after I went to offline mode. Looks like the reason is behind onyx update. I will investigate futher on this topic, but it might be not so easy to solve.

**Comment by CortneyOfstad** [comment]
Sounds good @Piotrfj — thanks!

**Comment by CortneyOfstad** [comment]
@Piotrfj do you know a rough ETA/plan as to when you think you'll have a PR — beginning of next week? EOW?

Thanks!

**Comment by Piotrfj** [comment]
Current status for assignment change:
My current guess is, as mentioned above, an onyx update. After going back to previous screen (tapping back) pusher action is fired (and in some cases even multiple ones). Scrolling on offline mode does not occur.
By the way, I've noticed that the keyboard is not always being shown (at least on my android emulator) despite of correct focus on composer, that might be something to take a look at, however this is not related to this issue.

**Comment by CortneyOfstad** [comment]
Sounds good @Piotrfj — thank you!

**Comment by koko57** [comment]
@CortneyOfstad Piotr meant that while we go back to the previous chat Onyx is updated and this update causes scrolling of the screen. When we go back while online GetNewerActions request is sent and the report data in Onyx is updated. 

I've checked it on the WEB and it also gets scrolled - let me know if I reproduced it correctly. I also tried when offline and it also gets scrolled after the data from the request is applied. It happens when you hover over the chat navigating back.

https://github.com/Expensify/App/assets/28020445/2a68bade-7ce9-455a-b6a2-68af7b285745

FYI: Piotr has been reassigned to another project from Callstack - me or someone from our team will take it over shortly and continue the investigation

**Comment by koko57** [comment]
@CortneyOfstad I'm continuing Piotr's investigation and what I've noticed is that removing `shouldEnableAutoScrollToTopThreshold` prop the list is not scrolled when going back. This https://github.com/Expensify/App/blob/4654d6ed3389916e750ecdfdbdad08d637535b8d/src/pages/home/report/ReportActionsView.tsx#L487 is always `true`, so the list is autoscrolled to the bottom. Looks like `hasNewestReportAction` condition is always `true` as well (I wonder if there are cases that is not true). For me doesn't really makes sense as I think it should be true only right after we get a new unread message, after we read it it should be false. 

I'll continue my investigation and think of a solution to suppress scrolling down when we don't get any new message.

**Comment by koko57** [comment]
@CortneyOfstad sorry, I've been busy with my other high priority issue, I haven't had much time to work on this one since Monday. I will look at it today. 

I think the root cause I've presented is correct, I don't think that there's another reason for this behaviour, so the only thing that I'm still thinking of is how to fix this issue.

**Comment by koko57** [comment]
Still looking for a proper solution

**Comment by CortneyOfstad** [comment]
@koko57 — any update on any solutions?

**Comment by CortneyOfstad** [comment]
Sounds good @koko57 👍

**Comment by CortneyOfstad** [comment]
@koko57 can you provide an update by EOD on where we're at with this? Thanks!

**Comment by koko57** [comment]
@CortneyOfstad I'm testing a few options, but still I don't quite understand the thing I mentioned [here](https://github.com/Expensify/App/issues/34573#issuecomment-2082829360). Is there anybody who can help me with this one?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by koko57** [comment]
@CortneyOfstad I merged with main today - and I can't reproduce it on Android either. The page still scrolls on the web. If it's desired for the web I think we can close this issue

**Comment by CortneyOfstad** [comment]
Thanks @koko57! I agree and we can close this. Thanks!
